src/rust/src/backend/poly1305.rs

```diff
@@ -7,26 +7,78 @@ use crate::buf::CffiBuf;
 use crate::error::{CryptographyError, CryptographyResult};
 use crate::exceptions;
 
-#[pyo3::prelude::pyclass(module = "cryptography.hazmat.bindings._rust.openssl.poly1305")]
-struct Poly1305 {
+#[cfg(any(CRYPTOGRAPHY_IS_BORINGSSL, CRYPTOGRAPHY_IS_LIBRESSL))]
+struct Poly1305Boring {
+    context: Option<Box<openssl_sys::poly1305_state>>,
```

Comment:
We need to seperate out the safe-wrapper part from the Python API part. Basically, the wrapper part should live in the cryptography-openssl crate -- see the hmac there as an example.

Reply:
Done! Now the safe wrapper lives in `cryptography-openssl`. I also added a small comment explaining the use of `Box` for the C struct data